lib/src/unit.dart

```diff
@@ -0,0 +1,17 @@
+import 'package:meta/meta.dart';
+
+/// Used instead of `void` as a return statement for a function
+/// when no value is to be returned.
+///
+/// There is only one value of type [Unit].
+@sealed
+class Unit {
+  const Unit._();
+
+  @override
+  String toString() => 'Unit';
+}
+
+/// Used instead of `void` as a return statement for a function when
+///  no value is to be returned.
+const unit = Unit._();
```

Comment:
We do have the `SuccessResult` with `success` global const [here](https://github.com/higorlapa/result/blob/main/lib/multiple_result.dart#L171-L186)
Do you still think it would be worth to have the `Unit` type? 

If yes, we'd have to deprecate the `SuccessResult`

Reply:
Yes because we need the error value if there is an error.
imagine the following situation:
Result<Unit, MyException> deleteInfo();

if delete return Unit, but I need the error if there is an exception.

Reply:
Got it, can you deprecate the `SuccessResult` then?

Reply:
I think so.